packages/react-textarea/Spec.md

```diff
@@ -2,62 +2,174 @@
+
+#### TextArea in v8
 
-- _Link to Open UI research_
-- _Link to comparison of v7 and v0_
-- _Link to GitHub epic issue for the converged component_
+- `Textfield` component with `multiline` property. When the `multiline` property is set to true, it is rendered as a `textarea`.
+- Has descriptions, label, error message, and handles maximum character count.
+
+#### TextArea in v0
+
+- Doesn't have `label` prop.
+- Handles maximum character count.
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+```tsx
+<TextArea
+  className="rootClassName"
+  style={{ background: 'blue' }}
+  id="textarea-1"
+  defaultValue="This is a TextArea"
+  onChange={(ev, data) => console.log(data.value)}
+  textArea={{ className: 'textAreaClassName', style: { color: 'green' } }}
+/>
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+- TextArea has six variants:
+  - `Filled Darker`: TextArea has a gray background and no outline.
+  - `Filled Lighter`: TextArea has a white background and no outline.
+  - `Underline`: TextArea has a gray background and a line on the bottom.
+  - `Outline`: TextArea has a white background, outline, and a line on the bottom.
+  - `Disabled`: TextArea has a gray background, gray text, outline, and interaction is disabled.
+  - `Read Only`: TextArea has a gray background, gray outline, and typing is disabled.
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+### Component props:
+
+```ts
+type TextAreaProps = {
+  /**
+   * Display inline instead of block.
+   *
+   * @defaultvalue false
+   */
+  inline?: boolean;
+
+  /**
+   * Which direction the TextArea is allowed to be resized.
+   *
+   * @defaultvalue off
+   */
+  resize?: 'off' | 'horizontal' | 'vertical' | 'both';
+
+  /**
+   * Direction the TextArea should resize.
+   *
+   * @defaultvalue off
+   */
+  autoResize?: 'off' | 'top' | 'bottom';
+
+  /**
+   * Size of the TextArea.
+   *
+   * @defaultvalue medium
+   */
+  size?: 'small' | 'medium' | 'large';
+
+  /**
+   * Styling the TextArea should use.
+   *
+   * @defaultvalue outline
+   */
+  appearance?: 'outline' | 'underline' | 'filledDarker' | 'filledLighter';
+
+  /**
+   * The value of the TextArea.
+   */
+  value?: string;
+
+  /**
+   * The default value of the TextArea.
+   */
+  defaultValue?: string;
+
+  /**
+   * Callback for when the user changes the value.
+   */
+  onChange?: (ev: React.FormEvent<HTMLTextAreaElement>, data: { value: string }) => void;
+};
+```
+
+As with most form controls, `TextArea` will have the `textarea` element as the primary element a wrapper (root). This means that `ref` will be passed directly to the `textarea` with its native props.
```

Comment:
Guessing this is what you meant?
```suggestion
As with most form controls, `TextArea` will have the `textarea` element as the primary slot (with a wrapper `root` element for styling). This means that `ref` will be passed directly to the `textarea` with its native props.
```

Though for TextArea, since it doesn't have contentBefore/contentAfter like Input, it might be possible to have the textarea itself as the root element--unless we're concerned about that creating limitations with future features we might want to add to the core component that could require a wrapper.

Reply:
The primary vs root slot thing is likely to be a point of confusion for users but I believe the pattern is used in all our input controls so it might be worth using the pattern for all input controls as a rule for the sake of consistency.

Reply:
After working in the implementation, I realized that it will need a root wrapper since `<textarea />` cannot have children. The design spec uses a custom focus indicator(the same one as input) that will require and `::after` pseudo-element and will not show because of this reason. I've updated the Spec to show the wrapper but let me know if you have any other ideas, I tried a few things but nothing seems to work.